algebird-test/src/main/scala/com/twitter/algebird/BaseProperties.scala

```diff
@@ -53,10 +53,8 @@ object BaseProperties {
 
   def isAssociative[T: Semigroup: Arbitrary] = isAssociativeDifferentTypes[T, T]
 
-  def semigroupSumWorks[T: Semigroup: Arbitrary: Equiv] = 'semigroupSumWorks |: forAll { (in: List[T]) =>
-    in.isEmpty || {
-      Equiv[T].equiv(Semigroup.sumOption(in.iterator).get, in.reduceLeft(Semigroup.plus(_, _)))
-    }
+  def semigroupSumWorks[T: Semigroup: Arbitrary: Equiv] = 'semigroupSumWorks |: forAll { (head: T, tail: List[T]) =>
+    Equiv[T].equiv(Semigroup.sumOption(head :: tail).get, tail.foldLeft(head)(Semigroup.plus(_, _)))
```

Comment:
actually, this is not such a great test, we should probably do:
```scala
Semigroup.sumOption(list.iterator) match {
  case None => list.isEmpty
  case Some(r) => Equiv[T].equiv(r, in.reduce(Semigroup.plus(_, _)))
}
```
but that is an actual change to the law, so we probably should not do it here.